parsons/community/community.py

```diff
@@ -0,0 +1,103 @@
+
+
+class Community(object):
+    """
+    Instantiate class.
+
+       `Args:`
+            community_client_id: str
+                The Community provided Client ID. Not required if ``COMMUNITY_CLIENT_ID`` env
+                variable set.
+            community_access_token: str
+                The Community provided access token. Not required if ``COMMUNITY_ACCESS_TOKEN`` env
+                variable set.
+            community_uri: str
+                The URI to access the CSV API. Not required, default is
+                https://secure.actblue.com/api/v1. You can set an ``ACTBLUE_URI`` env variable or
+                use this URI parameter if a different endpoint is necessary - for example, when
+                running this code in a test environment where you don't want to hit the actual API.
+    """
+
+    def __init__(self, community_client_id=None, community_access_token=None, community_url=None):
+        self.community_client_id = check_env.check("community_client_id", community_client_id)
+        self.community_access_token = check_env.check(
+            "community_access_token", community_access_token
+        )
+        self.uri = (
+            check_env.check("COMMUNITY_URL", community_url, optional=True)
+            or f"{COMMUNITY_API_ENDPOINT}/{community_client_id}/"
+        )
+        self.headers = {
+            "Authorization": f"Bearer {self.community_access_token}",
+            # "accept": "*/*",
+            # "accept-encoding": "gzip",
+        }
+        self.client = APIConnector(
+            self.uri,
+            headers=self.headers,
+        )
+
+    def get_request(self, resource):
+        """
+        GET request to Community.com API to get the CSV data.
+
+        `Args:`
+            resource: str
+                Data resource you are requesting.
+                Options:
+                    'campaigns': Campaign Performance data
+                    'outbound_message_type_usage`: Message Segment Usage data
+                    'campaign_links': Campaign Link Performance data
+                    'members': Member Details data
+                    'member_state_changes': Member Subscription Status data
+                    'custom_member_data': Custom Member Data
+                    'communities': Communities data
+                    'member_communities': Member Communities data
+
+        `Returns:`
+            Response of GET request; a successful response returns the CSV formatted data
+        """
+
+        logger.info(f"Requesting {resource}")
+        url = (
+            f"{resource}.csv.gz"
+            if resource != "outbound_message_type_usage"
+            else f"{resource}.csv.gz/segment-based-subscription"
+        )
+        response = self.client.get_request(url=url)
+        return response
+
+    def get_resource(self, resource):
+        """
+        Get specified data from Community.com API as Parsons table.
+
+        `Args:`
+            resource: str
```

Comment:
Can you rename the parameter and update the docs so we're using `file_name` instead of `resource`? As we chatted about on our call, this will make it easier for people to understand the `file_name error` response. Please also include a link to the underlying com api docs.